### ISSUE TITLE
[java] UnusedFormalParameter false-positive with qualified super ctor call

### ISSUE THREAD

**Issue by linusjf** [pmd/pmd#2937]
**Affects PMD Version:** 
6.29.0

**Description:**

False positive for rule [UnusedFormalParameter](https://pmd.github.io/latest/pmd_rules_java_bestpractices.html#unusedformalparameter) when constructor parameter is enclosing class instance used to invoke super constructor of enclosing class' child in constructor of subclass of its inner (child) class.

**Code Sample demonstrating the issue:**

```java

package jls;

public final class SuperClassTest {

  private SuperClassTest() {
    throw new IllegalStateException("Private constructor.");
  }

  public static void main(String[] args) {
    Outer x = new Outer();
    ChildOfInner a = new ChildOfInner(x);
    ChildOfInner b = new ChildOfInner(x);
    System.out.println(b.getSecret());
    a.setSecret(6);
    System.out.println(b.getSecret());
  }
}

class Outer {
  int secret = 5;

  class Inner {
    int getSecret() {
      return secret;
    }

    void setSecret(int s) {
      secret = s;
    }
  }
}

class ChildOfInner extends Outer.Inner {
  ChildOfInner(Outer x) {
    x.super();
  }
}

```
Example from JLS 15.

**Running PMD through:** *[CLI | Ant | Maven | Gradle | Designer | Other]*

**Comment by jsotuyod** [comment]
This is no longer reproducible under PMD 7.0.0
